### ISSUE TITLE
libp2p not building at 0.37.1

### ISSUE THREAD

**Issue by apopiak** [libp2p/rust-libp2p#2231]
### Problem Description
Version `0.37.1` is not currently building because of failed cargo dependency resolution. (This breaks a lot of dependents, e.g. Substrate.)

Note: The error below is different from the one triggered in when trying to build Substrate. Substrate fails to select a version for `snow`.

### Reproduction Steps
+ check out libp2p `git checkout v.0.37.1`
+ `cargo check`

#### Error Logs
```
error: failed to select a version for the requirement `aesni = "^0.7"`
candidate versions found which didn't match: 0.99.99, 0.10.0, 0.9.0, ...
location searched: crates.io index
required by package `aes v0.4.0`
    ... which is depended on by `aes-gcm v0.6.0`
    ... which is depended on by `snow v0.7.1`
    ... which is depended on by `libp2p-noise v0.30.0 (/home/alexander/dev/misc/rust-libp2p/transports/noise)`
    ... which is depended on by `libp2p v0.37.1 (/home/alexander/dev/misc/rust-libp2p)`
    ... which is depended on by `libp2p-relay v0.2.0 (/home/alexander/dev/misc/rust-libp2p/protocols/relay)`
```

**Comment by mxinden** [comment]
Reason for the compile time error:

- `libp2p` `v0.37.1` depends on `snow` `v0.7.1`.
- `snow` `v0.7.1` depends on `aes-gcm` `v0.6.0`.
- `aes-gcm` `v0.6.0` depends on `aes` `v0.4.0`. 
- `aes` `v0.4.0` depends on `aesni`. 
- [`aesni` has been deprecated](https://docs.rs/crate/aesni/0.99.99) with `v0.7` removed.

Fix:

- `libp2p` `v0.39.1` depends on `snow` `v0.8.0`.
- `snow` `v0.8.0` depends on `aes-gcm` `v0.9.0`.
- `aes-gcm` `v0.9.0` depends on `aes` `v0.7.5`.
- `aes` `v0.7.5` does **not** depend on `aesni`.

Would using `libp2p` `v0.39.1` be an option for you @apopiak?

**Comment by apopiak** [comment]
Yes and no. More recent versions of Substrate upgraded to 0.39 already, but the people in the ecosystem who depend on older versions are stuck with 0.37.
If it's possible and not too hard, releasing a 0.37.2 would be amazing.

**Comment by mxinden** [comment]
>  If it's possible and not too hard, releasing a 0.37.2 would be amazing.

`libp2p-noise` `v0.30.1` would be enough, no?

I have created branch [`libp2p-noise-0.30`](https://github.com/libp2p/rust-libp2p/tree/libp2p-noise-0.30) which is currently at tag `libp2p-noise-0.30.0` (the failing version). @apopiak could you create a pull request with:

- A bump of the `libp2p-noise` version to `v0.30.1`.
- A bump of `snow` in `libp2p-noise` to `v0.8.0`.
- The required changes to make the upgrade compile.

I can then merge the pull request and cut a new release of `libp2p-noise` based on the branch.

**Comment by Milerius** [comment]
@mxinden Do you think it's possible to have backward compatibility of #2264 with https://github.com/libp2p/rust-libp2p/commit/20183c1ea152f5bfe183543e4934e082c1428011 if possible

**Comment by apopiak** [comment]
> > If it's possible and not too hard, releasing a 0.37.2 would be amazing.
> 
> `libp2p-noise` `v0.30.1` would be enough, no?
> 
> I have created branch [`libp2p-noise-0.30`](https://github.com/libp2p/rust-libp2p/tree/libp2p-noise-0.30) which is currently at tag `libp2p-noise-0.30.0` (the failing version). @apopiak could you create a pull request with:
> 
>     * A bump of the `libp2p-noise` version to `v0.30.1`.
> 
>     * A bump of `snow` in `libp2p-noise` to `v0.8.0`.
> 
>     * The required changes to make the upgrade compile.
> 
> 
> I can then merge the pull request and cut a new release of `libp2p-noise` based on the branch.

will try to do this

**Comment by mxinden** [comment]
`libp2p-noise` `v0.30.1` has been [tagged](https://github.com/libp2p/rust-libp2p/tree/libp2p-noise-0.30.1) and [released](https://crates.io/crates/libp2p-noise/0.30.1).
